pkg/util/http.go

```diff
@@ -235,3 +245,24 @@ func SerializeProtoResponse(w http.ResponseWriter, resp proto.Message, compressi
 	}
 	return nil
 }
+
+type MaxBytesHandler struct {
```

Comment:
Using handler for switching `r.Body` to `http.MaxBytesReader` is too disconnected from where we actually read the body. Let's use more direct code.

Reply:
I feel the other way round. Having it at handler level allows us to not think about handling the size limit everywhere (if we miss somewhere, potential attack surface?). For example, we use the upstream Prometheus alertmanager which takes the http request and we don't have control over it.